File: JS and TypeScript/B59 Spiral Matrix 2.js
const n=6;
const arr= Array.from(Array(n), _ => Array(n).fill(0));
var i=0;
var cnt = 1;
const LIM =Math.floor((n+1)/2);
for (let loop = 0; loop<LIM;loop++){
    // iterate from left to right
    for(i =loop;i<n-loop;i++) arr[loop][i] = cnt++;
    // iterate from top to bottom 
    for(i =loop+1;i<n-loop;i++) arr[i][n - loop - 1] = cnt++;
    // iterate from right to left
    for(i =n-loop-2;i>loop-1;i--) arr[n - loop - 1][i] = cnt++;
    // iterate from bottom to top
    for(i =n-loop-2;i>loop;i--) arr[i][loop] = cnt++;
};
for (let j=0;j<n;j++){
    console.log(arr[j]);
}

File: Python/B59 Spiral Matrix 2.py
n= int(input("So chieu matran: "))
ans = [[0] * n for _ in range(n)]
cnt = 1
for loop in range((n + 1) // 2):
    # direction 1 - traverse from left to right
    for i in range(loop, n - loop):
        ans[loop][i] = cnt
        cnt += 1
    # # direction 2 - traverse from top to bottom
    for i in range(loop + 1, n - loop):
        ans[i][n - loop - 1] = cnt
        cnt += 1
    # # direction 3 - traverse from right to left
    for i in range(n - loop - 2, loop - 1, -1):
        ans[n - loop - 1][i] = cnt
        cnt += 1
    # # direction 4 - traverse from bottom to top
    for i in range(n - loop - 2, loop, -1):
        ans[i][loop] = cnt
        cnt += 1
    
for i in range(n):
    print(ans[i])